main/src/ca/uwaterloo/flix/language/phase/typer/TypeReduction.scala

```diff
@@ -322,6 +322,17 @@ object TypeReduction {
       val arrType = phantomArr.getClass
       arrType
 
+    // Vectors
+    case Type.Apply(Type.Cst(TypeConstructor.Vector, _), elmType, _) =>
+      // Find the Java class of the element type.
+      val t = getJavaType(elmType)
+
+      // Find the Java class of the array type.
+      // See: https://stackoverflow.com/questions/1679421/how-to-get-the-array-class-for-a-given-class-in-java
```

Comment:
This is a gammeldags solution. Class arrayType is more direct. Maybe we can go through all instances of this in another PR

Reply:
See if you can get it to work. I could not!